encoding/test/main.go

```diff
@@ -49,30 +50,55 @@ func readpoints() {
 }
 
 func TestKzgRs() {
-	numSymbols := 3
+	isSmallTest := false
+
+	numSymbols := 4096 / 8
 	// encode parameters
-	numNode := uint64(4) // 200
-	numSys := uint64(2)  // 180
+	numNode := uint64(4096) // 200
+	numSys := uint64(512)   // 180
 	numPar := numNode - numSys
-	// Prepare data
-	fmt.Printf("* Task Starts\n")
-	fmt.Printf("    Num Sys: %v\n", numSys)
-	fmt.Printf("    Num Par: %v\n", numPar)
-	//fmt.Printf("    Data size(byte): %v\n", len(inputBytes))
+
+	numDevices, _ := cr.GetDeviceCount()
+	fmt.Println("num device ", numDevices)
 
 	kzgConfig := &kzg.KzgConfig{
-		G1Path:          "g1.point",
-		G2Path:          "g2.point",
+		G1Path:          "../../inabox/resources/kzg/g1.point.600000",
+		G2Path:          "../../inabox/resources/kzg/g2.point.600000",
 		CacheDir:        "SRSTables",
-		SRSOrder:        3000,
-		SRSNumberToLoad: 3000,
+		SRSOrder:        300000,
+		SRSNumberToLoad: 300000,
 		NumWorker:       uint64(runtime.GOMAXPROCS(0)),
+		Verbose:         true,
+	}
+
+	if isSmallTest {
+		numSymbols = 2
+		numNode = 4
+		numSys = 2
+		numPar = numNode - numSys
+		kzgConfig = &kzg.KzgConfig{
+			G1Path:          "../../inabox/resources/kzg/g1.point.300000",
+			G2Path:          "../../inabox/resources/kzg/g2.point.300000",
+			CacheDir:        "SRSTables",
+			SRSOrder:        3000,
+			SRSNumberToLoad: 3000,
+			NumWorker:       uint64(runtime.GOMAXPROCS(0)),
```

Comment:
```suggestion
			NumWorker:       uint64(runtime.NumCPU()),
```
Think you're just trying to get the number of CPUs here right? Since GOMAXPROCS(0) doesn't have any effect except returning the current set value. A more explicit way to do this is runtime.NumCPU()